### PR TITLE
ADS: Line items 2/3 - List Item Switch

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL:  https://app.asana.com/0/0/1202923922178324/f

### Description

This PR Implement the new Switch theming

- Adds a new Component that represents Switches
- Create Lint rules that prevent use of SwitchCompat and SwitchMaterial
- Update all switches in the app with the new component

### Steps to test this PR
Usual smoke tests

### CODE REVIEW

File: app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/feature/settings/NetworkSwitchSettingPluginTestView.kt [hunk 1]
```diff
@@ -25,7 +25,7 @@ import org.junit.Before
 import org.junit.Test
 import org.mockito.kotlin.*
 
-class NetworkSwitchSettingPluginTest {
+class NetworkSwitchSettingPluginTestView {
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/test/java/com/duckduckgo/mobile/android/vpn/feature/settings/NetworkSwitchSettingPluginTestView.kt:28]: I see this was renamed from `NetworkSwitchSettingPluginTest` to `NetworkSwitchSettingPluginTestView`, but I think it should be `NetworkSwitchingSettingPluginTest` (third version :smiley:).

File: app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/VpnInternalSettingsActivity.kt [hunk 2]
```diff
@@ -188,9 +189,12 @@ class VpnInternalSettingsActivity : DuckDuckGoActivity() {
                 binding.debugLoggingToggle.isEnabled = isEnabled
                 binding.transparencyModeToggle.isEnabled = isEnabled
                 // only show this toggle when the variant is correct
-                binding.vpnNewNetworkingLayerToggle.isVisible = vpnNetworkStackVariantManager.getVariant() == "ng"
-                binding.vpnNewNetworkingLayerToggle.isEnabled = isEnabled
-                binding.settingsInfo.isVisible = !isEnabled
+                binding.vpnNewNetworkingLayerToggle.isEnabled = vpnNetworkStackVariantManager.getVariant() == "ng"
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/VpnInternalSettingsActivity.kt:192]: The toggle should be visible for the NG networking layer and hidden for the legacy one and enabled based on the value of `isEnabled`. The below lines are needed:
```
binding.vpnNewNetworkingLayerToggle.isVisible = vpnNetworkStackVariantManager.getVariant() == "ng"
binding.vpnNewNetworkingLayerToggle.isEnabled = isEnabled
```

File: app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/VpnInternalSettingsActivity.kt [hunk 3]
```diff
@@ -253,30 +257,28 @@ class VpnInternalSettingsActivity : DuckDuckGoActivity() {
...
+        binding.badHealthMitigationToggle.setIsChecked(appTpConfig.isEnabled(AppTpSetting.BadHealthMitigation))
         binding.badHealthMitigationToggle.setOnCheckedChangeListener(badHealthMitigationFeatureToggle)
     }
 
     private fun setupConfigSection() {
         with(AppTpSetting.Ipv6Support) {
-            binding.ipv6SupportToggle.isChecked = appTpConfig.isEnabled(this)
-            binding.ipv6SupportToggle.setOnCheckedChangeListener { _, isChecked ->
+            binding.ipv6SupportToggle.setIsChecked(appTpConfig.isEnabled(this))
+            binding.ipv6SupportToggle.setOnCheckedChangeListener { view, isChecked ->
```
Comment (anikiki) [inline app-tracking-protection/vpn-internal/src/main/java/com/duckduckgo/vpn/internal/feature/VpnInternalSettingsActivity.kt:281]: `view` can be replaced by `_`

File: autoconsent/autoconsent-impl/src/main/res/layout/activity_autoconsent_settings.xml [hunk 4]
```diff
@@ -15,34 +15,31 @@
...
-        <androidx.appcompat.widget.SwitchCompat
-            android:id="@+id/autoconsentToggle"
-            style="@style/SettingsSwitch"
-            android:paddingStart="16dp"
-            android:paddingEnd="16dp"
-            android:text="@string/autoconsentToggle"
-            android:theme="@style/SettingsSwitchTheme" />
+    <com.duckduckgo.mobile.android.ui.view.listitem.OneLineListItemSwitch
+        android:id="@+id/autoconsentToggle"
+        app:primaryText="@string/autoconsentToggle"/>
```
Comment (anikiki) [inline autoconsent/autoconsent-impl/src/main/res/layout/activity_autoconsent_settings.xml:43]: Needs height and width:
```
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
```

File: common-ui/src/main/res/layout/row_two_line_item.xml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?><!--
-  ~ Copyright (c) 2021 DuckDuckGo
+  ~ Copyright (c) 2018 DuckDuckGo
```
Comment (anikiki) [inline common-ui/src/main/res/layout/row_two_line_item.xml:2]: Back in 2018 :smile:

Comment (anikiki) [reply common-ui/src/main/res/layout/row_two_line_item.xml:2]: Not sure if it's related to this change, but I noticed the downloads are now missing the leading / trailing icon.
downloads_miss_leading_trailing_icons

File: app/src/main/res/layout/content_privacy_dashboard.xml [hunk 6]
```diff
@@ -148,10 +148,11 @@
                 app:layout_constraintTop_toTopOf="parent" />
 
             <androidx.appcompat.widget.SwitchCompat
+                tools:ignore="AndroidSwitchInXml"
                 android:id="@+id/privacyToggle"
                 android:layout_width="wrap_content"
                 android:layout_height="26dp"
-                android:theme="@style/SwitchTheme"
+                android:theme="@style/Widget.DuckDuckGo.Switch.White"
```
Comment (anikiki) [inline app/src/main/res/layout/content_privacy_dashboard.xml:155]: Looks like this toggle lost its styling (is now blue).
blue_toggle

File: app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksAdapter.kt [hunk 7]
```diff
@@ -193,17 +193,17 @@ sealed class BookmarkScreenViewHolders(itemView: View) : RecyclerView.ViewHolder
...
             )
-            twoListItem.setTitle(bookmark.title)
-            twoListItem.setSubtitle(parseDisplayUrl(bookmark.url))
-            loadFavicon(bookmark.url, twoListItem.imageView())
+            twoListItem.setPrimaryText(bookmark.title)
+            twoListItem.setSecondaryText(parseDisplayUrl(bookmark.url))
+            loadFavicon(bookmark.url, twoListItem.leadingIcon())
 
-            twoListItem.setOverflowClickListener { anchor ->
+            twoListItem.setTrailingIconClickListener { anchor ->
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksAdapter.kt:206]: I noticed the bookmarks / favorites don't have leading / trailing icons anymore.
bookmarks_no_icons

Comment (malmstein) [reply app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksAdapter.kt:206]: its visibility was hidden if the values were not passed through xml. It's fixed now :D

File: app/src/main/res/layout/content_privacy_dashboard.xml (hunk 6 above)
Comment (malmstein) [reply app/src/main/res/layout/content_privacy_dashboard.xml:155]: nice catch, I'll revert!

**Review by anikiki — APPROVED** [review]
Looks good! :tada:

**Pull request merged**